File: src/main/java/org/tensorflow/data/TensorFrame.java
package org.tensorflow.data;

import org.tensorflow.Operand;
import org.tensorflow.Session;
import org.tensorflow.op.Ops;
import org.tensorflow.types.family.TType;

public abstract class TensorFrame<T extends TType> implements Dataset<T>, GraphLoader<T> {
  protected long batchSize = 1;
  protected boolean dropRemainder = false;

  /* Override functions from Dataset<T> */
  @Override
  public Dataset<T> batch(long batchSize, boolean dropRemainder) {
    this.batchSize = batchSize;
    this.dropRemainder = dropRemainder;
    return this;
  }

  public long batchSize() {
    return this.batchSize;
  }

  public long numBatches() {
    return size() / batchSize;
  }
}

File: src/main/scala/org/tensorflow/data/TensorFrame.scala
package org.tensorflow.data

import org.tensorflow.types.family.TType

abstract class TensorFrame[T <: TType] extends Dataset[T] with GraphLoader[T] {
  protected var _batchSize = 1L
  protected var dropRemainder = false

  /* Override functions from Dataset<T> */ override def batch(batchSize: Long, dropRemainder: Boolean): Dataset[T] = {
    _batchSize = batchSize
    this.dropRemainder = dropRemainder
    this
  }

  override def batchSize: Long = _batchSize

  override def numBatches: Long = size / _batchSize
}
